core/src/main/kotlin/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractorV2.kt

```diff
@@ -225,10 +227,26 @@ fun makeSimpleReportTrain(
             return@rdp abs(point.speed - projSpeed) * speedScaling +
                 abs(point.time - projTime) * timeScaling
         }
+
+    // Check if the scheduled points were respected
+    var scheduledPointsHonored = true
+    for (e in schedule) {
+        if (e.arrival == null) continue
+        var computed = envelopeStopWrapper.interpolateTotalTimeUS(e.pathOffset.distance.meters)
+        if (e.stopFor != null) computed -= e.stopFor.milliseconds * 1_000
+        val expected = e.arrival.milliseconds * 1_000
+        // Check that the expected arrival time and the simulated one match ~1s
+        if ((computed - expected).absoluteValue > 1_000_000) {
+            scheduledPointsHonored = false
+            break
+        }
+    }
```

Comment:
Nit: could be moved to a separate function for better readability (and control flow)